src/actions/vstudio/vs2010_vcxproj.lua

```diff
@@ -1231,7 +1231,15 @@
 
 
 	function m.generateDebugInformation(cfg)
-		m.element("GenerateDebugInformation", nil, tostring(cfg.flags.Symbols ~= nil))
+		if cfg.flags.Symbols and _ACTION >= "vs2015" then
+			if cfg.flags.FastLink then
+				m.element('GenerateDebugInformation', nil, "DebugFastLink")
+			else
+				m.element('GenerateDebugInformation', nil, "Debug")
```

Comment:
What @samsinsane is saying.... I'm all for a proper API, but the problem is, much of this only applies to VS2015, so the "symbols" api (or whatever the name is) should only allow certain values for certain actions... and there currently does not exist such a feature in premake.